### PR TITLE
Migrating to go modules

### PR DESCRIPTION
**Pull request by InTheCloudDan**
Set it using go 1.12 because some of the dependencies had build constraints requiring that version as modules and rather than fight them went with 1.12 because by the time it becomes default we will probably be using that version or higher.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,11 +62,27 @@ container-uninstall:
...
+	make tidy
+else
 godep:
 	dep ensure
+endif
+
+update:
+	$(GO_BIN) get -u
+
+tidy:
```
Comment (lumjjb) [inline Makefile]: if it's only used in the case of `GO111MODULE`, let's move this into the `if` block above.

File: Makefile [hunk 2]
```diff
@@ -61,11 +62,27 @@ container-uninstall:
 	sudo hack/update_binaries.sh delete
 
 .PHONY: godep
+ifeq ($(GO111MODULE),on)
+godep:
+	$(GO_BIN) build -v ./...
```
Comment (lumjjb) [inline Makefile]: this is just the dep step, building shouldn't happen in here?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module github.com/nabla-containers/runnc
+
+go 1.12
```
Comment (lumjjb) [inline go.mod]: This feels a little troubling.. is there a reason for locking in a golang version here? Anyway to do a version range?

File: .travis.yml [hunk 4]
```diff
@@ -12,8 +12,14 @@ go:
   - tip
 
 matrix:
+  include:
+    - go: "1.12"
+      env: GO111MODULE=on
+    - go: tip
+      env: GO111MODULE=on
```
Comment (lumjjb) [inline .travis.yml]: can we add a check in the matrix for older versions not using `GO111MODULE=on` (and using `dep ensure`) until we reach go 1.13, where gomodules will be standard? When that happens we will remove all the older dep stuff including the `Gopkg.toml/lock`

File: Makefile (hunk 1 above)
Comment (InTheCloudDan) [reply Makefile]: I'll double check might have been messed up with my rebase.

File: go.mod (hunk 3 above)
Comment (InTheCloudDan) [reply go.mod]: I'll look into the version lock of go mod. 1.11 didn't add it for some reason but building with 1.12 does.

File: .travis.yml (hunk 4 above)
Comment (InTheCloudDan) [reply .travis.yml]: with this setup it runs 4 builds. 2 without the `GO111MODULE` and 2 with it on.

Comment (lumjjb) [reply .travis.yml]: Ok sounds good, this is just me not being familiar with jenkins then.

File: go.mod (hunk 3 above)
Comment (InTheCloudDan) [reply go.mod]: I checked on gopher slack, this version currently does not have any impact on the build. It's there for future use cases.

File: Makefile (hunk 2 above)
Comment (InTheCloudDan) [reply Makefile]: fixed!

**Pull request merged**